### DISCUSSION TITLE
Laser lines in opposite directions do not match

### DISCUSSION THREAD

**Discussion by Vcoderlab** [grblHAL/core#578] [General]
Hi everyone! I am trying to do laser engraving and I found that the lines made by moving the laser in opposite directions do not match.

I wrote a test G-CODE file and am attaching the file itself and the results of its work. Blue lines in the image show the direction of the laser movement. Speed of movement is 1000 mm/min.

![Laser](https://github.com/user-attachments/assets/cb43213d-2ffc-4e07-9bf0-c2a96263b22e)

[Laser on-off speed test 1000.zip](https://github.com/user-attachments/files/16824927/Laser.on-off.speed.test.1000.zip)

I am using GrblHAL with BTT SKR Pico v1.0 board. But when I tried the same G-CODE with BTT SKR 3 EZ board, I got the same result. So I think it is not the driver issue.

What could be the problem? How can this discrepancy be eliminated or compensated for?

**Comment by terjeio** [comment]
> What could be the problem?

Mechanical [backlash](https://duckduckgo.com/?t=ffab&q=cnc+backlash&ia=web).

> How can this discrepancy be eliminated or compensated for?

Best approach is to fix it mechanically, tighten belts, fasten grub screws properly etc.
Alternatively it can be done in firmware, [enable backlash compensation](https://github.com/grblHAL/core/blob/7725fab5b8901c666259e010675d3259b4c8491b/config.h#L501), rebuild and reflash, then adjust $160 (X) and $161 (Y) until the backlash is cancelled out. It can also be enabled in the [Web Builder](http://svn.io-engineering.com:8080/) in the _Advanced features_ tab.

**Reply by Vcoderlab** [reply]
Thanks a lot for the advice! It really looks like mechanical backlash. Although the drive belts are tight and do not sag. Using the parameters $160 and $161 I was able to compensate it.
